src/dirac_cwl_proto/production/__init__.py

```diff
@@ -166,24 +175,27 @@ def _get_transformations(
         step_task = _create_subworkflow(
             step, str(production.task.cwlVersion), production.task.inputs
         )
-        query_params = _get_query_params(production.task)
+        configuration = _get_configuration(production.task)
 
-        # Get the metadata & description for the step
+        # Get the execution_hooks & description for the step
         step_id = step.id.split("#")[-1]
-        step_data: ProductionStepMetadataModel = production.steps_metadata.get(
+        step_data: TransformationExecutionHooksHint | ExecutionHooksHint = (
```

Comment:
Can it be an `ExecutionHooksHint`?
```suggestion
        step_data: TransformationExecutionHooksHint = (
```